conanfile.txt

```diff
@@ -0,0 +1,9 @@
+[requires]
+libelfin/[>=0.3]
```

Comment:
I like the idea of allowing the use of conan as it makes it very convenient for folks that just want to reproduce the results of the published paper or shown in one of the talks, or just want to play with the benchmarks. However, currently the use of conan isn't just possible, it is mandatory as libelfin itself does not provide a CMake config file and there is no find module shipped with either CMake itself or coz and thus the `find_package(libelfin)` call will fail unless used with conan (or someone invests the time to write their own config file). This leads to issues in a few situations:

- coz is available as a system package in ubuntu. Distro maintainers need complete control over the used libs and compilation flags used to build these libs. It is absolutely necessary that the version of coz packaged for ubuntu uses the system libelfin version. So requiring the use of conan makes the life of the package maintainer much more difficult. They need to manually patch the buildsystem to remove the conan stuff and find a way to still get coz to build.
-  A user of a distribution that does not have system packages might still want to install coz globally. What happens, if that user happens to have another application/library installed that does make use of libelfin itself and thus there already is a system libelfin binary installed? Should they overwrite their system libelfin with the newer version that comes with conan? What if these aren't binary compatible? Even if they install coz in a different location (say /opt), currently it is not guaranteed that libcoz.so will pick up a libelfin.so placed next to it instead of the system libelfin.so because we don't set an RPATH.

There are at least two options here that I can think of:

- Remove the use of conan entirely and don't use `find_package` to find libelfin. Add `elfin` to the target link libraries and don't detect the missing library at configure time and let the user figure out the problem when the linker yells at them. Note that for the benchmarks you can still use `find_package` as CMake ships find modules for sqlite and bzip2 since version 3.14. You could even make them optional dependencies and add the respective benchmark only if the required lib has been found so that someone that is only interested in, say, the linear algebra benchmark doesn't first need to install sqlite.
- Write a find module for libelfin that can be used if there is no conan-created config file. You'd need to ensure that the conan config files are preferred over the find module though, which is possible by setting the `CMAKE_FIND_PACKAGE_PREFER_CONFIG` that was introduced in CMake 3.15. Alternatively you could first search for libelfin in config mode with `find_package(libelfin CONFIG QUIET)` and if it isn't found try again with `find_package(libelfin MODULE REQUIRED)` if you don't like to increase the minimum required CMake version to 3.15. For the benchmarks you could again use the existing CMake find modules.

The first option has the advantage that it isn't too much effort. You do lose the benefit of detecting the missing library at configure time though. It also makes it more difficult to build coz against a version of libelfin that isn't installed globally as the user now needs to explicitly modify the `CMAKE_MODULE_LINK_FLAGS` to include the linker search path. The second option is more user-friendly, but writing a robust find module is pretty difficult unless libelfin ships pkg-config files, in which case the `FindPkgConfig` module can be used. Bonus points if the find module also detects the version number so that the `find_package` call can also abort the build if the found libelfin version is too old.

Reply:
This makes it work with original libelfin pkg-config package without Conan libelfin package.
https://github.com/plasma-umass/coz/pull/178/commits/f1a7978f82a3d7b1a20d015e3480685cc85ce7de